### PR TITLE
OCPBUGS-18984: pkg/payload/precondition/clusterversion/etcdbackup: Drop precondition

### PR DESCRIPTION
**Pull request by wking**
We'd added this precondition back in ee2dcd13e5 (#637), [to prepare for etcd moving to 3.5.0 in OpenShift 4.9][1].  Since then, openshift/cluster-etcd-operator#835 started taking backups around every update, not just minor-version updates.  Those backups [may be taken after an update is accepted by the cluster-version operator][1.5], at which point restoring from backup and not updating might be difficult, but the [coming `TechPreviewNoUpgrade`-in-4.14 backup controller][2] from openshift/cluster-etcd-operator#1066 may mitigate that if it moves towards stabilization.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1997347#c0
[1.5]: https://issues.redhat.com/browse/OCPBUGS-18984
[2]: https://github.com/openshift/api/blob/15f6221a9976f6bbb9a7103d16bf7e2618e55567/config/v1/types_feature.go#L183

### CODE REVIEW

**Review by geliu2016 — APPROVED** [review]
/label cherry-pick-approved

**Comment by Elbehery** [comment]
lint fails on 
```
kg/payload/precondition/clusterversion/upgradeable.go:147:6: func `minorVersionUpgrade` is unused (unused)
func minorVersionUpgrade(currentMinor string, desiredMinor string) bool {
     ^
```

File: pkg/payload/precondition/clusterversion/upgradeable.go [hunk 1]
```diff
@@ -87,8 +87,8 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 
 	// if there is no difference in the minor version (4.y.z where 4.y is the same for current and desired), then we can still upgrade
 	// if no cluster overrides have been set
-	if currentMinor == desiredMinor {
-		klog.V(2).Infof("Precondition %q passed: minor from the current %s matches minor from the target %s (both %s).", pf.Name(), currentVersion, releaseContext.DesiredVersion, currentMinor)
+	if !minorVersionUpgrade(currentMinor, desiredMinor) {
+		klog.V(2).Infof("Precondition %q passed: minor from the target %s is not a minor version update from the current %s.", pf.Name(), releaseContext.DesiredVersion, currentVersion, currentMinor)
```
Comment (wking) [inline pkg/payload/precondition/clusterversion/upgradeable.go:91]: > ...func `minorVersionUpgrade` is unused...

I've resolved this lint complaint by pivoting `Upgradeable=False` to use this function, as described in e17c9648

**Pull request merged**